### ISSUE TITLE
Apache client throws exception for BASIC auth in POST

### ISSUE THREAD

**Issue by cjstehno** [http-builder-ng/http-builder-ng#10]
In the `HttpPostSpec` test, the `'[#client] POST (BASIC) /basic: returns content'` test fails for the Apache client but passes for the Java client. The exception thrown is:

```
java.lang.RuntimeException: org.apache.http.client.ClientProtocolException
    at groovyx.net.http.optional.ApacheHttpBuilder.exec(ApacheHttpBuilder.java:292)
    at groovyx.net.http.optional.ApacheHttpBuilder.doPost(ApacheHttpBuilder.java:347)
    at groovyx.net.http.HttpObjectConfigImpl.nullInterceptor(HttpObjectConfigImpl.java:39)
    at groovyx.net.http.HttpBuilder.post(HttpBuilder.java:693)
    at groovyx.net.http.HttpPostSpec.[#client] POST (BASIC) /basic: returns content(HttpPostSpec.groovy:165)
Caused by: org.apache.http.client.ClientProtocolException
    at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:186)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:220)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:164)
    at groovyx.net.http.optional.ApacheHttpBuilder.exec(ApacheHttpBuilder.java:289)
    ... 4 more
Caused by: org.apache.http.client.NonRepeatableRequestException: Cannot retry request with a non-repeatable request entity.
```

which seems related to this SO post (http://stackoverflow.com/questions/8222929/httppost-failed-due-to-cannot-retry-request-with-a-non-repeatable-request-entit) about repeatable entities. Our HttpEntity implementation is "not" repeatable, but I am not sure whether that is intentional or just "because".

**Comment by dwclark** [comment]
They can be repeatable, the ApacheHttpBuilder class would have to buffer the InputStreams and only throw them away once the response starts. I punted on this when implementing it hoping it wouldn't be a problem.

The correct solution is to implement that buffered input stream, storing the contents in an array of `ByteBuffer` up to a maximum limit configured `HttpObjectConfig.Client` or `HttpObjectConfig.Execution`. Default out of the box would be something reasonable like 4 MB. Once 4 MB had been buffered the buffer would signal to the Apache HttpClient that the request is no longer repeatable and stop buffering. In 99.9% of the cases, the client won't be uploading large amounts of data and this is acceptable.

I'm guessing that the Java HttpUrlConnection is doing this internally by default.

**Comment by cjstehno** [comment]
This seems like a minor issue. I should be able to take this one up once I am done with testing/documenting the verbs if you don't pick it up first.

**Comment by cjstehno** [comment]
I have opened #56 - just noting as a related issue.

**Comment by cjstehno** [comment]
Issue seems to be test related and is resolved in my ersatz testing branch - closing this for now.

**Comment by cjstehno** [comment]
Doh! Nope, this is still an issue... it seems to fail during full suite test runs.

**Comment by cjstehno** [comment]
This has been fixed in development and will be merged with the 0.14.0 release.

**Comment by cjstehno** [comment]
This has been addressed in the 0.14.0 release (development branch)
